packages/core/src/controllers/verify.ts

```diff
@@ -1,76 +1,120 @@
   private isDevEnv;
-  // the queue is only used during the loading phase of the iframe to ensure all attestations are posted
-  private queue: string[] = [];
-  // flag to disable verify when the iframe fails to load on main & fallback urls.
-  // this means Verify API is not enabled for the current projectId and there's no point in trying to initialize it again.
-  private verifyDisabled = false;
-
-  constructor(public projectId: string, public logger: Logger) {
-    super(projectId, logger);
+  private verifyUrlV2 = VERIFY_SERVER_V2;
+  private publicKey?: jwk;
+
+  constructor(public projectId: string, public logger: Logger, public store: IKeyValueStorage) {
+    super(projectId, logger, store);
     this.logger = generateChildLogger(logger, this.name);
-    this.verifyUrl = VERIFY_SERVER;
     this.abortController = new AbortController();
     this.isDevEnv = isNode() && process.env.IS_VITEST;
+    this.init();
   }
 
-  public init: IVerify["init"] = async (params) => {
-    if (this.verifyDisabled) return;
-
-    // ignore on non browser environments
-    if (isReactNative() || !isBrowser()) return;
-
-    const verifyUrl = this.getVerifyUrl(params?.verifyUrl);
-    // if init is called again with a different url, remove the iframe and start over
-    if (this.verifyUrl !== verifyUrl) {
-      this.removeIframe();
-    }
-    this.verifyUrl = verifyUrl;
+  get storeKey(): string {
+    return `verify:public:key`;
+  }
 
-    try {
-      await this.createIframe();
-    } catch (error) {
-      this.logger.info(`Verify iframe failed to load: ${this.verifyUrl}`);
-      this.logger.info(error);
-      // if the iframe fails to load, disable verify
-      this.verifyDisabled = true;
+  public init = async () => {
+    this.publicKey = await this.store.getItem(this.storeKey);
+    if (this.publicKey && toMiliseconds(this.publicKey?.expiresAt) < Date.now()) {
+      this.logger.debug("verify v2 public key expired");
+      await this.removePublicKey();
     }
   };
 
   public register: IVerify["register"] = async (params) => {
-    if (!this.initialized) {
-      this.addToQueue(params.attestationId);
-      await this.init();
-    } else {
-      this.sendPost(params.attestationId);
+    if (!isBrowser()) return;
+    const { id, decryptedId } = params;
+    const url = `${this.verifyUrlV2}/attestation?projectId=${this.projectId}`;
+    let src = "";
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify({ id, decryptedId }),
+      });
+      const { srcdoc } = await response.json();
+      src = srcdoc;
+      this.logger.debug("srcdoc fetched", src);
+    } catch (e) {
+      return;
     }
+    const document = getDocument() as Document;
+    const abortTimeout = this.startAbortTimer(ONE_SECOND * 2);
+    const attestatiatonJwt = await new Promise((resolve) => {
+      const abortListener = () => {
+        document.body.removeChild(iframe);
```

Comment:
Do we need to reject the promise here?

Reply:
started rejecting so we can diff between success/failure cases